repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py

```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
                             Policies.Retry.Phase)
         flags = Flags(request_restart_after_phase=True)
 
     class InitRamStartPhase(Phase):
+        '''Reboot into the next system therefore not a real step, however the
+        before and post parts are interesting to have for hooking.
+        '''
         name = 'InitRamStart'
         filter = TagFilter(InitRamStartPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class NetworkPhase(Phase):
+        '''Bring up the network.'''
         name = 'Network'
         filter = TagFilter(NetworkPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class StoragePhase(Phase):
+        '''Mount all storage points.'''
         name = 'Storage'
         filter = TagFilter(StoragePhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class LateTestsPhase(Phase):
+        '''Tests that need to be executed later.'''
         name = 'LateTests'
         filter = TagFilter(LateTestsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class PreparationPhase(Phase):
+        '''YUM Configurations. Backup not upgradable configurations (No Verify
+        Configs).
+        '''
         name = 'Preparation'
         filter = TagFilter(PreparationPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class RPMUpgradePhase(Phase):
+        '''Upgrade RPMs. Check all files are installed (troubles with handling
+        dirs/symlinks), includes reinstallation of affected packages.
+        '''
         name = 'RPMUpgrade'
         filter = TagFilter(RPMUpgradePhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ApplicationsPhase(Phase):
+        '''Application of changes on configuration files, move various
+        directores, etc.
+        '''
         name = 'Applications'
         filter = TagFilter(ApplicationsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ThirdPartyApplicationsPhase(Phase):
+        '''A place for custom actors for third party applications to be
+        upgraded.
+        '''
         name = 'ThirdPartyApplications'
         filter = TagFilter(ThirdPartyApplicationsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class FinalizationPhase(Phase):
+        '''Plan SELinux relabeling. Labels Bootloader settings.'''
```

Comment:
Actually, we probably will do SELinux relabeling during this phase in future. What about: "Rest of actions that should be done before the reboot into the ugpraded system. E.g. selinux relabeling, ...."